### ISSUE TITLE
SQL warning and errors on glpi 10.0.10 relate to inventory

### ISSUE THREAD

**Issue by ftoledo** [glpi-project/glpi#15724]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.10

### Bug description

first error is related to devicebattery and second are when i add time intervals to run inventory tasks

```
cat sql-errors.log

[2023-10-05 17:08:17] glpisqllog.WARNING: DBmysql::query() in /html/src/DBmysql.php line 419
  *** MySQL query warnings:
  SQL: SELECT * FROM `glpi_lockedfields` WHERE `itemtype` = 'Item_DeviceBattery' AND ((`items_id` = '' OR `is_global` = '1'))
  Warnings: 
1292: Truncated incorrect DECIMAL value: ''
  Backtrace :
  src/DBmysqlIterator.php:112                        
  src/DBmysql.php:1078                               DBmysqlIterator->execute()
  src/Lockedfield.php:185                            DBmysql->request()
  src/Lockedfield.php:158                            Lockedfield->getLocks()
  src/Inventory/Asset/InventoryAsset.php:433         Lockedfield->getLockedNames()
  src/Inventory/Asset/Device.php:175                 Glpi\Inventory\Asset\InventoryAsset->handleInput()
  src/Inventory/Asset/MainAsset.php:937              Glpi\Inventory\Asset\Device->handle()
  src/Inventory/Asset/MainAsset.php:852              Glpi\Inventory\Asset\MainAsset->handleAssets()
  src/RuleImportAsset.php:969                        Glpi\Inventory\Asset\MainAsset->rulepassed()
  src/Rule.php:1533                                  RuleImportAsset->executeActions()
  src/RuleCollection.php:1640                        Rule->process()
  src/Inventory/Asset/MainAsset.php:574              RuleCollection->processAllRules()
  src/Inventory/Inventory.php:722                    Glpi\Inventory\Asset\MainAsset->handle()
  src/Inventory/Inventory.php:353                    Glpi\Inventory\Inventory->handleItem()
  src/Inventory/Request.php:360                      Glpi\Inventory\Inventory->doInventory()
  src/Inventory/Request.php:90                       Glpi\Inventory\Request->inventory()
  src/Agent/Communication/AbstractRequest.php:359    Glpi\Inventory\Request->handleAction()
  src/Agent/Communication/AbstractRequest.php:271    Glpi\Agent\Communication\AbstractRequest->handleJSONRequest()
  front/inventory.php:94                             Glpi\Agent\Communication\AbstractRequest->handleRequest()
  ...tplace/glpiinventory/front/communication.php:72 include_once()
  marketplace/glpiinventory/index.php:45             include_once()
  public/index.php:82                                require()
  {"user":"inventory@tom"} 
[2023-10-05 22:31:57] glpisqllog.ERROR: DBmysql::query() in /html/src/DBmysql.php line 379
  *** MySQL query error:
  SQL: INSERT INTO `glpi_plugin_glpiinventory_timeslotentries` (`plugin_glpiinventory_timeslots_id`, `day`, `begin`, `end`) VALUES ('3', NULL, '0', '86400')
  Error: Column 'day' cannot be null
  Backtrace :
  src/DBmysql.php:1349                               
  src/CommonDBTM.php:722                             DBmysql->insert()
  src/CommonDBTM.php:1323                            CommonDBTM->addToDB()
  ...e/glpiinventory/inc/timeslotentry.class.php:413 CommonDBTM->add()
  ...e/glpiinventory/front/timeslotentry.form.php:55 PluginGlpiinventoryTimeslotEntry->addEntry()
  public/index.php:82                                require()
  {"user":"70@tom"} 
tom:/home/glpi.pressenter.com.ar/html/files/_log# 
tom:/home/glpi.pressenter.com.ar/html/files/_log# cat sql-errors.log
[2023-10-05 17:08:17] glpisqllog.WARNING: DBmysql::query() in /html/src/DBmysql.php line 419
  *** MySQL query warnings:
  SQL: SELECT * FROM `glpi_lockedfields` WHERE `itemtype` = 'Item_DeviceBattery' AND ((`items_id` = '' OR `is_global` = '1'))
  Warnings: 
1292: Truncated incorrect DECIMAL value: ''
  Backtrace :
  src/DBmysqlIterator.php:112                        
  src/DBmysql.php:1078                               DBmysqlIterator->execute()
  src/Lockedfield.php:185                            DBmysql->request()
  src/Lockedfield.php:158                            Lockedfield->getLocks()
  src/Inventory/Asset/InventoryAsset.php:433         Lockedfield->getLockedNames()
  src/Inventory/Asset/Device.php:175                 Glpi\Inventory\Asset\InventoryAsset->handleInput()
  src/Inventory/Asset/MainAsset.php:937              Glpi\Inventory\Asset\Device->handle()
  src/Inventory/Asset/MainAsset.php:852              Glpi\Inventory\Asset\MainAsset->handleAssets()
  src/RuleImportAsset.php:969                        Glpi\Inventory\Asset\MainAsset->rulepassed()
  src/Rule.php:1533                                  RuleImportAsset->executeActions()
  src/RuleCollection.php:1640                        Rule->process()
  src/Inventory/Asset/MainAsset.php:574              RuleCollection->processAllRules()
  src/Inventory/Inventory.php:722                    Glpi\Inventory\Asset\MainAsset->handle()
  src/Inventory/Inventory.php:353                    Glpi\Inventory\Inventory->handleItem()
  src/Inventory/Request.php:360                      Glpi\Inventory\Inventory->doInventory()
  src/Inventory/Request.php:90                       Glpi\Inventory\Request->inventory()
  src/Agent/Communication/AbstractRequest.php:359    Glpi\Inventory\Request->handleAction()
  src/Agent/Communication/AbstractRequest.php:271    Glpi\Agent\Communication\AbstractRequest->handleJSONRequest()
  front/inventory.php:94                             Glpi\Agent\Communication\AbstractRequest->handleRequest()
  ...tplace/glpiinventory/front/communication.php:72 include_once()
  marketplace/glpiinventory/index.php:45             include_once()
  public/index.php:82                                require()
  {"user":"inventory@tom"} 
[2023-10-05 22:31:57] glpisqllog.ERROR: DBmysql::query() in /html/src/DBmysql.php line 379
  *** MySQL query error:
  SQL: INSERT INTO `glpi_plugin_glpiinventory_timeslotentries` (`plugin_glpiinventory_timeslots_id`, `day`, `begin`, `end`) VALUES ('3', NULL, '0', '86400')
  Error: Column 'day' cannot be null
  Backtrace :
  src/DBmysql.php:1349                               
  src/CommonDBTM.php:722                             DBmysql->insert()
  src/CommonDBTM.php:1323                            CommonDBTM->addToDB()
  ...e/glpiinventory/inc/timeslotentry.class.php:413 CommonDBTM->add()
  ...e/glpiinventory/front/timeslotentry.form.php:55 PluginGlpiinventoryTimeslotEntry->addEntry()
  public/index.php:82                                require()
  {"user":"70@tom"} 

cat php-errors.log

[2023-10-05 22:33:28] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "lastday" in /html/marketplace/glpiinventory/inc/timeslotentry.class.php at line 285
  Backtrace :
  ...e/glpiinventory/front/timeslotentry.form.php:55 PluginGlpiinventoryTimeslotEntry->addEntry()
  public/index.php:82                                require()
  
[2023-10-05 22:33:28] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "beginday" in /html/marketplace/glpiinventory/inc/timeslotentry.class.php at line 285
  Backtrace :
  ...e/glpiinventory/front/timeslotentry.form.php:55 PluginGlpiinventoryTimeslotEntry->addEntry()
  public/index.php:82                                require()
  
[2023-10-05 22:33:28] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "lastday" in /html/marketplace/glpiinventory/inc/timeslotentry.class.php at line 288
  Backtrace :
  ...e/glpiinventory/front/timeslotentry.form.php:55 PluginGlpiinventoryTimeslotEntry->addEntry()
  public/index.php:82                                require()
  
[2023-10-05 22:33:28] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "beginday" in /html/marketplace/glpiinventory/inc/timeslotentry.class.php at line 288
  Backtrace :
  ...e/glpiinventory/front/timeslotentry.form.php:55 PluginGlpiinventoryTimeslotEntry->addEntry()
  public/index.php:82                                require()
  
[2023-10-05 22:33:28] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "lasthours" in /html/marketplace/glpiinventory/inc/timeslotentry.class.php at line 289
  Backtrace :
  ...e/glpiinventory/front/timeslotentry.form.php:55 PluginGlpiinventoryTimeslotEntry->addEntry()
  public/index.php:82                                require()
  
[2023-10-05 22:33:28] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "beginhours" in /html/marketplace/glpiinventory/inc/timeslotentry.class.php at line 289
  Backtrace :
  ...e/glpiinventory/front/timeslotentry.form.php:55 PluginGlpiinventoryTimeslotEntry->addEntry()
  public/index.php:82                                require()
```

### Your GLPI setup information

glpi 10.0.10 tarball / debian 10 / php 8.1

**Comment by trasher** [comment]
Error related to inventory task concerns glpiinventory plugin, please open a bug on relevant repository.

For the lock field issue, I have no idea how to reproduce; maybe can you provide more information.

**Comment by stonebuzz** [comment]
Error related to inventory task seems indicated a bad configuration for timeslot

It must be checked

**Comment by trasher** [comment]
Also, you didi not provide requested informaiton so we do not know if you use other plugins; but if it's the case, please tests without any other plugin.

**Comment by cedric-anne** [comment]
The following error has been fixed by #15746:

```
  *** MySQL query warnings:
  SQL: SELECT * FROM `glpi_lockedfields` WHERE `itemtype` = 'Item_DeviceBattery' AND ((`items_id` = '' OR `is_global` = '1'))
  Warnings: 
1292: Truncated incorrect DECIMAL value: ''
```

Other errors are located in the `glpiinventory` plugin itself. Please open an issue for them on the [plugin repository](https://github.com/glpi-project/glpi-inventory-plugin/issues).
